src/tests/integration/conftest.py

```diff
@@ -223,3 +266,36 @@ def ip_with_MSSQL(ip_empty, setup_MSSQL):
     yield ip_empty
     # Disconnect database
     ip_empty.run_cell("%sql -x " + alias)
+
+
+@pytest.fixture(scope="session")
+def setup_Snowflake(test_table_name_dict):
```

Comment:
I thought about it, was more about the technical difficulty 

1. I understand what you mean, like generalizing the `setup_*`. But I couldn't find the way to define the dynamic fixture in the running time, in each test case we need to use `request.getfixturevalue` to parse each parametrized ip_with_dynamic_db 
2. Also, in each `setup_[Database]`, we will need to use with context to initialize the different container environments, I think to prevent the Overgeneralization, separating those should be fine now

But meanwhile I will still keep looking if anyway to generalize this